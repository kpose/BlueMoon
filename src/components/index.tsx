import Button from './Button/Button';
import InventoryCard from './InventoryCard/InventoryCard';
import DeleteModal from './DeleteModal/DeleteModal';
import Fab from './FAB/Fab';

export {Button, InventoryCard, DeleteModal, Fab};
