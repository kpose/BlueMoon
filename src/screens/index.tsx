import Home from './Home/Home';
import CreateInventory from './CreateInventory/CreateInventory';
import EditInventory from './EditInventory/EditInventory';

export {Home, CreateInventory, EditInventory};
