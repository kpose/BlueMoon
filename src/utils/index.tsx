import * as sizes from './fonts';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from './responsive';
import * as colors from './colors';
import {createInventory, getInventory, deleteInventory} from './asyncInventory';

export {sizes, wp, hp, colors, createInventory, getInventory, deleteInventory};
