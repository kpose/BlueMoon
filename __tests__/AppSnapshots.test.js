import * as React from 'react';
import renderer from 'react-test-renderer';
import {render, cleanup} from '@testing-library/react-native';
import {Home, EditInventory, CreateInventory} from '../src/screens';

afterEach(cleanup);

describe('Snapshot text of different screens', () => {
  const {} = render(<Home />);

  const mockedParams = {
    route: {params: {refresh: Function}},
    navigation: '',
  };

  it('home screen renders correctly', () => {
    const tree = renderer.create(<Home />);
    expect(tree).toMatchSnapshot();
  });

  it('Create screen renders correctly', () => {
    const tree = renderer.create(<CreateInventory {...mockedParams} />);
    expect(tree).toMatchSnapshot();
  });

  it('Edit screen renders correctly', () => {
    const tree = renderer.create(<EditInventory {...mockedParams} />);
    expect(tree).toMatchSnapshot();
  });
});
