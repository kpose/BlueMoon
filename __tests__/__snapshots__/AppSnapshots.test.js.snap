// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot text of different screens Create screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#8c94a1",
        "borderRadius": 15,
        "elevation": 133.5,
        "height": 934,
        "marginBottom": 26.5,
        "marginTop": 13.5,
        "width": 712.5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Enter Name"
      placeholderTextColor="#2A9D8F"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgb(0,0,0)",
            "borderRadius": 22.5,
            "color": "rgb(255,255,255)",
            "height": 93.5,
            "marginTop": 26.5,
            "width": 600,
          },
          Object {
            "fontSize": 23,
          },
        ]
      }
      testID="nameID"
      textAlign="center"
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      keyboardType="number-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Total Stock"
      placeholderTextColor="#2A9D8F"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgb(0,0,0)",
            "borderRadius": 22.5,
            "color": "rgb(255,255,255)",
            "height": 93.5,
            "marginTop": 26.5,
            "width": 600,
          },
          Object {
            "fontSize": 23,
          },
        ]
      }
      testID="totalID"
      textAlign="center"
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      keyboardType="number-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Price"
      placeholderTextColor="#2A9D8F"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgb(0,0,0)",
            "borderRadius": 22.5,
            "color": "rgb(255,255,255)",
            "height": 93.5,
            "marginTop": 26.5,
            "width": 600,
          },
          Object {
            "fontSize": 23,
          },
        ]
      }
      testID="priceID"
      textAlign="center"
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Description"
      placeholderTextColor="#2A9D8F"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "rgb(0,0,0)",
          "borderRadius": 22.5,
          "color": "rgb(255,255,255)",
          "height": 267,
          "marginBottom": 53.5,
          "marginTop": 26.5,
          "width": 600,
        }
      }
      testID="descriptionID"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#2A9D8F",
          "borderRadius": 37.5,
          "height": 93.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": 375,
        }
      }
      testID="buttonID"
    >
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontSize": 17,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;

exports[`Snapshot text of different screens Edit screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 23,
        },
        Object {
          "color": "#2A9D8F",
          "fontWeight": "bold",
        },
      ]
    }
  >
     Inventory
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#8c94a1",
        "borderRadius": 15,
        "elevation": 133.5,
        "height": 934,
        "marginBottom": 26.5,
        "marginTop": 13.5,
        "width": 712.5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholderTextColor="#2A9D8F"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgb(0,0,0)",
            "borderRadius": 22.5,
            "color": "rgb(255,255,255)",
            "height": 93.5,
            "marginTop": 26.5,
            "width": 600,
          },
          Object {
            "fontSize": 23,
          },
        ]
      }
      textAlign="center"
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      keyboardType="number-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholderTextColor="#2A9D8F"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgb(0,0,0)",
            "borderRadius": 22.5,
            "color": "rgb(255,255,255)",
            "height": 93.5,
            "marginTop": 26.5,
            "width": 600,
          },
          Object {
            "fontSize": 23,
          },
        ]
      }
      textAlign="center"
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      keyboardType="number-pad"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholderTextColor="#2A9D8F"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "rgb(0,0,0)",
            "borderRadius": 22.5,
            "color": "rgb(255,255,255)",
            "height": 93.5,
            "marginTop": 26.5,
            "width": 600,
          },
          Object {
            "fontSize": 23,
          },
        ]
      }
      textAlign="center"
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholderTextColor="#2A9D8F"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "rgb(0,0,0)",
          "borderRadius": 22.5,
          "color": "rgb(255,255,255)",
          "height": 267,
          "marginBottom": 53.5,
          "marginTop": 26.5,
          "width": 600,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#2A9D8F",
          "borderRadius": 37.5,
          "height": 93.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": 375,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
            },
            Object {
              "fontSize": 17,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Update
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#F84751",
              "fontWeight": "bold",
              "marginTop": 40,
            },
            Object {
              "fontSize": 18,
              "letterSpacing": -0.08,
            },
          ]
        }
      >
        Delete this Inventory
      </Text>
    </View>
  </View>
</View>
`;

exports[`Snapshot text of different screens home screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Text>
    Welcome to inventory
  </Text>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2A9D8F",
        "borderRadius": 100,
        "bottom": 66.5,
        "height": 80,
        "justifyContent": "center",
        "position": "absolute",
        "right": 52.5,
        "width": 187.5,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 17,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Create
    </Text>
  </View>
</View>
`;
